meson.build

```diff
@@ -0,0 +1,74 @@
+project('lcm', 'c', 'cpp',
+  version : '1.5.0',
+  default_options : ['default_library=static',
+                     'buildtype=release',
+                     'c_std=gnu99',
+                     'warning_level=0',
+                     'c_args=-Wno-format-zero-length'])
+
+# Dependencies
+glib_dep = dependency('glib-2.0')
+
+# set windows dependencies
+unix = host_machine.system() != 'windows'
+windows = not unix
+if windows
+  cpp = meson.get_compiler('cpp')
+  wsock32 = cpp.find_library('wsock32')
+  ws2_32 = cpp.find_library('ws2_32')
+endif
+
+# Core modules
+# lcm-gen compiles with warnings that cannot be disabled when lcm is used
+# as a Meson subproject in a parent project with more warnings enabled. Allow
+# parent projects to disable compilation of lcm-gen to fix warnings as needed.
+subdir('lcm')
+subdir('lcm-logger')
+if get_option('lcm_enable_lcmgen').enabled()
+  subdir('lcmgen')
+endif
+
+# Documentation (Main, C/C++, .NET)
+subdir('docs')
+
+# Doesn't seem to compile examples but compiles tests?
+# We don't want to compile examples, which makes this option misleading, but it
+# is how it is in the CMakeLists.txt file.
+# Examples
+if get_option('lcm_enable_examples').enabled()
+  subdir('liblcm-test')
+endif
+
+# Python
+if get_option('lcm_enable_python').enabled()
+  subdir('lcm-python')
+endif
+
+# Java (Work in progress)
```

Comment:
It doesn't seem to compile, probably because I removed the gtest subproject, so I will simply comment out the entire thing so people can assume it's not working.